### PR TITLE
feat: restart runtime after alter system query

### PR DESCRIPTION
**Pull request by tolgadur**
### Description 
Once the system configs are altered we need to restart the runtimes to pick up the change. This PR is dependend on https://github.com/confluentinc/ksql/pull/8434

### Testing done 
unit tests 

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java [hunk 1]
```diff
@@ -213,6 +213,11 @@ public boolean hasActiveQueries() {
     return !primaryContext.getQueryRegistry().getPersistentQueries().isEmpty();
   }
 
+  public void restartStreamsRuntime() {
```
Comment (wcarlson5) [inline ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java:216]: Why do we need this at this level? would it be better to just call this whenever the alter system is called? Or should we leave them decoupled? WDYT

File: ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java [hunk 2]
```diff
@@ -221,6 +223,23 @@ public TransientQueryMetadata createStreamPullQuery(
     return query;
   }
 
+  @Override
+  public void restartStreamsRuntime(final KsqlConfig config) {
```
Comment (wcarlson5) [inline ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java:227]: Can you change the name to include this is picking up the new configs?

**Review by wcarlson5 — COMMENTED** [review]
its on its way

File: ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java (hunk 1 above)
Comment (tolgadur) [reply ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java]: I think we need to seperate it as we call the method in the `ValidatedCommandFactory` for validation purposes. If we didn't the streams runtime.

Comment (wcarlson5) [reply ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java:216]: ah yeah thats a good point

**Review by wcarlson5 — APPROVED** [review]
LGTM

**Pull request merged**